src/test/scala/testUtils/ExternalServicesTestUtils.scala

```diff
@@ -0,0 +1,259 @@
+import uk.gov.nationalarchives.dp.client.Entities.Entity
+import uk.gov.nationalarchives.dp.client.EntityClient
+import uk.gov.nationalarchives.{DADynamoDBClient, DASNSClient}
+import uk.gov.nationalarchives.dr2.Lambda
+import uk.gov.nationalarchives.dr2.Lambda.CompactEntity
+
+import java.time.ZonedDateTime
+import java.util.UUID
+
+class ExternalServicesTestUtils extends AnyFlatSpec with BeforeAndAfterEach with BeforeAndAfterAll {
+  val graphQlServerPort = 9001
+
+  val wiremockGraphqlServer = new WireMockServer(graphQlServerPort)
+
+  override def beforeAll(): Unit = {
+    wiremockGraphqlServer.start()
+  }
+
+  override def afterAll(): Unit = {
+    wiremockGraphqlServer.stop()
+  }
+
+  override def afterEach(): Unit = {
+    wiremockGraphqlServer.resetAll()
+  }
+
+  case class MockLambda(
+      entitiesUpdatedSinceReturnValue: List[IO[Seq[Entity]]] = List(
+        IO(
+          Seq(
+            Entity(
+              "CO",
+              UUID.fromString("4148ffe3-fffc-4252-9676-595c22b4fcd2"),
+              Some("test file1"),
+              deleted = false,
+              "path/1"
+            ),
+            Entity(
+              "IO",
+              UUID.fromString("7f094550-7af2-4dc3-a954-9cd7f5c25d7f"),
+              Some("test file2"),
+              deleted = false,
+              "path/2"
+            ),
+            Entity(
+              "SO",
+              UUID.fromString("d7879799-a7de-4aa6-8c7b-afced66a6c50"),
+              Some("test file3"),
+              deleted = false,
+              "path/3"
+            )
+          )
+        ),
+        IO(
+          Seq(
+            Entity(
+              "SO",
+              UUID.fromString("b10d021d-c013-48b1-90f9-e4ccc6149602"),
+              Some("test file4"),
+              deleted = false,
+              "path/4"
+            ),
+            Entity(
+              "IO",
+              UUID.fromString("e9f6182f-f1b4-4683-89be-9505f5c943ec"),
+              Some("test file5"),
+              deleted = false,
+              "path/5"
+            ),
+            Entity(
+              "CO",
+              UUID.fromString("97f49c11-3be4-4ffa-980d-e698d4faa52a"),
+              Some("test file6"),
+              deleted = false,
+              "path/6"
+            )
+          )
+        )
+      ),
+      entityEventActionsReturnValue: IO[Seq[EventAction]] = IO(
+        Seq(
+          EventAction(
+            UUID.fromString("f24313ce-dd5d-4b28-9ebc-b47893f55a8e"),
+            "Ingest",
+            ZonedDateTime.parse("2023-06-06T20:39:53.377170+01:00")
+          )
+        )
+      ),
+      getAttributeValuesReturnValue: IO[Map[String, AttributeValue]] = IO(
+        Map("datetime" -> AttributeValue.builder().s("2023-06-06T20:39:53.377170+01:00").build())
+      ),
+      snsPublishReturnValue: IO[List[PublishBatchResponse]] = IO(List(PublishBatchResponse.builder().build())),
+      updateAttributeValuesReturnValue: IO[Int] = IO(200)
+  ) extends Lambda() {
```

Comment:
This is really nice. We should definitely make this the standard for testing lambda functions.

Reply:
Thanks 🙂